pkgs/top-level/all-packages.nix

```diff
@@ -6366,7 +6366,11 @@ in
   waf = callPackage ../development/tools/build-managers/waf { python = python3; };
   wafHook = makeSetupHook {
     deps = [ python ];
-    substitutions = { inherit waf; };
+    substitutions = {
+      inherit waf;
+      crossFlags = lib.optionalString (stdenv.hostPlatform != stdenv.targetPlatform)
+        ''--cross-compile "--cross-execute=${stdenv.targetPlatform.emulator pkgs}"'';
```

Comment:
Note that this is pretty expensive to build. I'm not sure if waf is used enough for it to a problem though. I think it's okay since all waf scripts support it (not all need it though).

Reply:
If I understand correctly, this will only be built when one cross-builds a derivation using wafHook, so this is pretty optimal.
```
~/src/nixpkgs static-proot* 
$   nix-store -qR $(nix-instantiate . -A pkgsCross.raspberryPi.talloc) | grep qemu
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/7s9kxar0nh6d6nzri342ynpfpxniwq1b-fix-qemu-ga.patch
/nix/store/pgkjdxzpglzfj5hpjim1p1m8bipx9hmj-qemu-3.1.0.tar.bz2.drv
/nix/store/3pjswv7hpf0z4avx11dp16rb9wfwpppg-qemu-3.1.0.drv
                                                                                                                                                                                                       
~/src/nixpkgs static-proot* 
$   nix-store -qR $(nix-instantiate . -A talloc) | grep qemu 
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
```

Reply:
Yeah that's right. There might be a benefit to only adding it when you know a script will need to cross execute. But there's not a clean way to do that in Nixpkgs right now.